core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/stack/StackRequestValidator.java

```diff
@@ -179,5 +191,40 @@ private PlatformResourceRequestJson getRequestForEncryptionKeys(Long credentialI
         return request;
     }
 
+    private void checkResourceRequirementsIfBlueprintIsDatalakeReady(StackRequest stackRequest, ValidationResultBuilder validationBuilder) {
+        Blueprint blueprint = blueprintService.getByNameForWorkspaceId(stackRequest.getClusterRequest()
+                .getBlueprintName(), restRequestThreadLocalService.getRequestedWorkspaceId());
+        boolean sharedServiceReadyBlueprint = Optional.ofNullable((Boolean) blueprint.getTags().getMap().get("shared_services_ready")).orElse(false);
+        if (sharedServiceReadyBlueprint) {
+            Set<RdsType> rdsTypes = getGivenRdsTypes(stackRequest.getClusterRequest());
+            String rdsErrorMessageFormat = "For a Datalake cluster (since you have selected a datalake ready blueprint) you should provide at least one %s "
+                    + "rds/database configuration to the Cluster request";
+            if (!rdsTypes.contains(RdsType.HIVE)) {
+                validationBuilder.error(String.format(rdsErrorMessageFormat, "Hive"));
+            }
+            if (!rdsTypes.contains(RdsType.RANGER)) {
+                validationBuilder.error(String.format(rdsErrorMessageFormat, "Ranger"));
+            }
+            if (isLdapNotProvided(stackRequest.getClusterRequest())) {
+                validationBuilder.error("For a Datalake cluster (since you have selected a datalake ready blueprint) you should provide an "
```

Comment:
You could unify all the error messages here with something like:
"For a Datalake cluster the following configuration was missing: {LDAP, external HIVE Database}"

Reply:
I prefer detailed error (verification) error handling